payments-core/src/main/java/com/stripe/android/paymentsheet/injection/PaymentSheetViewModelModule.kt

```diff
@@ -0,0 +1,224 @@
+package com.stripe.android.paymentsheet.injection
+
+import android.app.Application
+import androidx.activity.result.ActivityResultCallback
+import com.stripe.android.Logger
+import com.stripe.android.PaymentConfiguration
+import com.stripe.android.PaymentController
+import com.stripe.android.PaymentRelayContract
+import com.stripe.android.StripeIntentResult
+import com.stripe.android.StripePaymentController
+import com.stripe.android.auth.PaymentBrowserAuthContract
+import com.stripe.android.model.StripeIntent
+import com.stripe.android.networking.ApiRequest
+import com.stripe.android.networking.StripeApiRepository
+import com.stripe.android.payments.DefaultReturnUrl
+import com.stripe.android.payments.PaymentFlowResult
+import com.stripe.android.payments.PaymentFlowResultProcessor
+import com.stripe.android.payments.PaymentIntentFlowResultProcessor
+import com.stripe.android.payments.SetupIntentFlowResultProcessor
+import com.stripe.android.payments.Stripe3ds2CompletionContract
+import com.stripe.android.paymentsheet.DefaultGooglePayRepository
+import com.stripe.android.paymentsheet.DefaultPrefsRepository
+import com.stripe.android.paymentsheet.GooglePayRepository
+import com.stripe.android.paymentsheet.PaymentSheetContract
+import com.stripe.android.paymentsheet.PaymentSheetViewModel
+import com.stripe.android.paymentsheet.PrefsRepository
+import com.stripe.android.paymentsheet.flowcontroller.ActivityLauncherFactory
+import com.stripe.android.paymentsheet.model.PaymentIntentClientSecret
+import com.stripe.android.paymentsheet.model.SetupIntentClientSecret
+import com.stripe.android.paymentsheet.repositories.PaymentMethodsApiRepository
+import com.stripe.android.paymentsheet.repositories.PaymentMethodsRepository
+import com.stripe.android.paymentsheet.repositories.StripeIntentRepository
+import dagger.Lazy
+import dagger.Module
+import dagger.Provides
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.first
+import javax.inject.Named
+import javax.inject.Provider
+import javax.inject.Singleton
+import kotlin.coroutines.CoroutineContext
+
+@Module
+internal class PaymentSheetViewModelModule {
```

Comment:
The shared deps of this Module and [FlowControlelrModule](https://github.com/stripe/stripe-android/blob/eac015357556989919e19b8ced7b43c3b4fec095/payments-core/src/main/java/com/stripe/android/paymentsheet/injection/FlowControllerModule.kt) will be merged into a single module in a follow up PR